TriforkSwiftExtensions/TriforkSwiftExtensions/Array/ArrayExtension.swift

```diff
@@ -42,4 +87,34 @@ public extension Array {
         a.append(newElement)
         return a
     }
+
+    /// Returns the first element of the `Array`.
+    var head: Element? { first }
+    
+    /// Removes the first element and retuns a new `Array` containing the rest of the elements.
+    var tail: [Element] { Array(dropFirst()) }
+    
+    /// Returns `nil` if the `Array` is empty, otherwise returning self.
+    var nilIfEmpty: [Element]? { isEmpty ? nil : self }
+    
+    /// Splits the `Array` into two dependent on the "shouldAppendToFirst-Argument", where the first `Array` contains the elements that match the bool and the second `Array` contains the rest.
+    func split(by shouldAppendToFirst: (Element) -> Bool) -> ([Element], [Element]) {
```

Comment:
only thought here is whenever one should make a struct for the result to name the first and second as to match the lambda argument.